contracts/examples/mintable-erc1155-transfer/FxMintableERC1155RootTunnel.sol

```diff
@@ -137,10 +146,24 @@ contract FxMintableERC1155RootTunnel is FxBaseRootTunnel, Create2, ERC1155Holder
         ) = abi.decode(syncData, (address, address, address, uint256[], uint256[], bytes, string));
         // if root token is not available, create it
         if (!Address.isContract(rootToken) && rootToChildTokens[rootToken] == address(0x0)) {
-            _deployRootToken(rootToken, metadata);
+            _deployRootToken(childToken, metadata);
         }
         require(rootToChildTokens[rootToken] == childToken, "FxMintableERC1155RootTunnel: INVALID_MAPPING_ON_EXIT");
-        IFxERC1155(rootToken).safeBatchTransferFrom(address(this), user, ids, amounts, data);
+        // check if current balance for token is less than amount,
+        // mint remaining amount for this address
+        IFxERC1155 nft = IFxERC1155(rootToken);
+        uint256 len = ids.length;
+        for (uint256 i; i < len; ) {
+            uint256 balance = nft.balanceOf(address(this), ids[i]);
+            if (balance < amounts[i]) {
+                nft.mint(address(this), ids[i], amounts[i] - balance, "");
```

Comment:
Could use `mintBatch` function to save gas. 
With the help of loop, find out the `ids` and `amounts` needed to mint, then call the `mintBatch` function with the array. Making a memory specific new array would be better than making external call multiple times.

Reply:
@DhairyaSethi please take a look when you're available.

Reply:
thank you! #61 has the recommendation. (not transferring directly to users because it might be complicate logic for indexers, would be great to know your thoughts)